app/presenters/tree_builder_configuration_manager.rb

```diff
@@ -66,19 +74,48 @@ def fetch_unassigned_configuration_profile_objects(count_only, configuration_man
     unassigned_configuration_profile_objs
   end
 
+  def fetch_unassigned_inventory_group_objects(count_only, configuration_manager_id)
+    no_inventory_configured_systems = ConfiguredSystem.where(:inventory_root_group_id => nil,
+                                                             :manager_id              => configuration_manager_id)
+    no_inventory_configured_systems_filtered = rbac_filtered_objects(no_inventory_configured_systems,
+                                                                     :match_via_descendants => ConfiguredSystem)
+    if no_inventory_configured_systems_filtered.count > 0
+      unassigned_id = "#{configuration_manager_id}-unassigned"
+      unassigned_inventory_group =
+        [ConfigurationProfile.new(:name       => "Unassigned Inventory Group|#{unassigned_id}",
+                                  :manager_id => configuration_manager_id)]
+      unassigned_inventory_group_objs = count_only_or_objects(count_only,
+                                                              unassigned_inventory_group,
+                                                              nil)
+    end
+
+    if unassigned_inventory_group_objs.nil?
+      count_only ? unassigned_inventory_group_objs = 0 : unassigned_inventory_group_objs = []
```

Comment:
Yes - in building the tress - returning the count to indicate the node has children and the actual children nodes when the node is expanded.